test/src/micro/attributes_test.dart

```diff
@@ -72,3 +90,55 @@ class AttributesTest extends PolymerElement {
 
   AttributesTest.created() : super.created();
 }
+
+enum Foobar { foo, bar }
+
+@PolymerRegister(_serializeAttributesName, hostAttributes: const {
+  'host-string': 'string',
+  'host-num': 2,
+  'host-bool': true,
+  //'host-map': const {'hello': 'world'},
+  //'host-list': const ['hello', 'world'],
+  // TODO(jakemac): Do we need to support this?
+//  'host-date-time': new DateTime(1987, 07, 18),
+  'host-foobar': Foobar.bar
+})
+class SerializedAttributesTest extends PolymerElement with PolymerSerialize {
```

Comment:
you could try extending AttributesTest, just to remove some boilerplate. Interestingly, I don't think polymer js allows you to extend elements, but I am pretty sure we can (although you will have to copy the dom-module if you have one). I haven't tried this yet, but I don't see why it wouldn't work.